api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/MetricsServiceHandler.java

```diff
@@ -66,34 +66,43 @@ protected String getTenant() {
         return new NamedDataPointObserver<>(response, mapper, type);
     }
 
-    <T> Observable<MetricId<T>> findMetricsByNameOrTag(List<String> metricNames, String tags, MetricType<T> type) {
+    <T> Observable<MetricId<T>> findMetricsByNameOrTag(List<String> metricNames, String tags, String tagsQuery,
+            MetricType<T> type) {
         List<String> nonNullNames = metricNames == null ? Collections.emptyList() : metricNames;
         Map<String, String> mapTags = TagsConverter.fromNullable(tags).getTags();
-        return findMetricsByNameOrTag(nonNullNames, mapTags, type);
+        return findMetricsByNameOrTag(nonNullNames, mapTags, tagsQuery, type);
     }
 
-    <T> Observable<MetricId<T>> findMetricsByNameOrTag(List<String> metricNames, Tags tags, MetricType<T> type) {
+    <T> Observable<MetricId<T>> findMetricsByNameOrTag(List<String> metricNames, Tags tags, String tagsQuery,
+            MetricType<T> type) {
         List<String> nonNullNames = metricNames == null ? Collections.emptyList() : metricNames;
         Map<String, String> mapTags = tags == null ? Collections.emptyMap() : tags.getTags();
-        return findMetricsByNameOrTag(nonNullNames, mapTags, type);
+        return findMetricsByNameOrTag(nonNullNames, mapTags, tagsQuery, type);
     }
 
     private <T> Observable<MetricId<T>> findMetricsByNameOrTag(@NotNull List<String> metricNames, @NotNull Map<String,
-            String> tags, MetricType<T> type) {
+    String> tags, String tagsQuery, MetricType<T> type) {
 
-        if (metricNames.isEmpty() && tags.isEmpty()) {
-            return Observable.error(new RuntimeApiError("Either metrics or tags parameter must be used"));
+        if (metricNames.isEmpty() && tags.isEmpty() && tagsQuery == null) {
+            return Observable.error(new RuntimeApiError("Either metrics or tags query parameters must be used"));
         }
-        if (!metricNames.isEmpty() && !tags.isEmpty()) {
-            return Observable.error(new RuntimeApiError("Cannot use both the metrics and tags parameters"));
+        if (!metricNames.isEmpty() && (!tags.isEmpty() || tagsQuery != null)) {
+            return Observable.error(new RuntimeApiError("Cannot use both the metrics and tags query parameters"));
+        }
+        if (metricNames.isEmpty() && (!tags.isEmpty() && tagsQuery != null)) {
```

Comment:
For the previous question, I guess it's catched here. But if it's a REST-interface limitation then it should be catched earlier.